### PR TITLE
[Azdevify] Change from rg deployment to sub deployment.

### PR DESCRIPTION
**Pull request by zedy-wj**
This PR updated the `azd` deployment from `rg` to `sub`, consistent with other repos.

**Update content:**

1. Remove the resource group imported from github action. (rg deployment is no longer used)
image

2. Store the `openai key` in the `KeyVault` and use sdk to obtain it in the source code.
image

3. Update infra/core.

@jongio for notification.

### CODE REVIEW

File: func/SessionProcessor.cs [hunk 1]
```diff
@@ -44,7 +46,20 @@ public static class SessionProcessor
 
         static SessionProcessor()
         {
-            var key = Environment.GetEnvironmentVariable("AzureOpenAI.Key");
+            var keyVaultEndpoint = Environment.GetEnvironmentVariable("AzureKeyVault.Endpoint");
```
Comment (jongio) [inline func/SessionProcessor.cs:49]: Could we switch this to use the capital and snake casing for environment variables like we do in the other samples?

**Review by jongio — CHANGES_REQUESTED** [review]

File: func/SessionProcessor.cs (hunk 1 above)
Comment (zedy-wj) [reply func/SessionProcessor.cs]: Updated.

**Comment by zedy-wj** [comment]
@jongio - Updated according to your comment. It's ready for re-review.

File: README.md [hunk 2]
```diff
@@ -85,23 +85,17 @@ Then download the template to the current directory:
 azd init -t Azure-Samples/azure-sql-db-session-recommender
```
Comment (jongio) [inline README.md:85]: Add a comment that they only need to provide the -t parameter if they haven't already cloned it.

File: .github/workflows/azure-dev.yml [hunk 3]
```diff
@@ -24,7 +24,6 @@ jobs:
       AZURE_TENANT_ID: ${{ vars.AZURE_TENANT_ID }}
```
Comment (jongio) [inline .github/workflows/azure-dev.yml:24]: Please look at the azure.yaml file to remove the post-provision hooks to see if the app still works.  

The goal should be to be able to run azd provision and then F5 in Visual Studio.

File: README.md (hunk 2 above)
Comment (jongio) [reply README.md:85]: Like we did here: https://github.com/jongio/chat-with-your-data-solution-accelerator/pull/2/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R131

Comment (zedy-wj) [reply README.md:85]: Updated.

File: .github/workflows/azure-dev.yml (hunk 3 above)
Comment (zedy-wj) [reply .github/workflows/azure-dev.yml:24]: This repo doesn't contain hooks in azure.yaml file, maybe that's what you want to say here? https://github.com/jongio/chat-with-your-data-solution-accelerator/pull/2/files#diff-1ba5bd52036276068bc787a1be3dfa49aa920317ab02fa460c99246a78e588f3R11-R17

Comment (jongio) [reply .github/workflows/azure-dev.yml:24]: I commented on the wrong project you can disregard.

**Comment by zedy-wj** [comment]
@jongio - This pr is ready for re-review.

**Comment by yorek** [comment]
I just tried the deployment. Deployment works well, but now I get the following error when running the Azure Function: 

```
The user, group or application 'appid=xxx;oid=yyy;iss=https://sts.windows.net/zzz/' does not have secrets get permission on key vault 'kv-bkcxyj6gqxrce;location=eastus2'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287","innererror":{"code":"AccessDenied"}"
```

**Comment by jongio** [comment]
@zedy-wj - can you please test this app with your updates to make sure it all works?

**Comment by zedy-wj** [comment]
@jongio , @yorek - We can't reproduce this issue, both deploying via `azd` and local are working fine at the moment.
image


@yorek - Can you provide more information? Including the steps to reproduce or even some executed commands? Thanks!
> I just tried the deployment. Deployment works well, but now I get the following error when running the Azure Function:
> 
> ```
> The user, group or application 'appid=xxx;oid=yyy;iss=https://sts.windows.net/zzz/' does not have secrets get permission on key vault 'kv-bkcxyj6gqxrce;location=eastus2'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287","innererror":{"code":"AccessDenied"}"
> ```

**Comment by yorek** [comment]
If just followed the instructions. Cloned the repo locally (via `git clone` not via `azd init` and then run `azd up`). I'm trying to deploy everything with `--debug` to see if I can get more details (and if the problem is still there).

**Comment by yorek** [comment]
> @jongio , @yorek - We can't reproduce this issue, both deploying via `azd` and local are working fine at the moment. image
> 
> @yorek - Can you provide more information? Including the steps to reproduce or even some executed commands? Thanks!
> 
> > I just tried the deployment. Deployment works well, but now I get the following error when running the Azure Function:
> > ```
> > The user, group or application 'appid=xxx;oid=yyy;iss=https://sts.windows.net/zzz/' does not have secrets get permission on key vault 'kv-bkcxyj6gqxrce;location=eastus2'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287","innererror":{"code":"AccessDenied"}"
> > ```

Have you tried running the SQL statement to add a row to the table? That is what triggers the function execution and then the error. Otherwise the function looks like it is working fine (as it is never called).

**Comment by zedy-wj** [comment]
@yorek - Thanks for your help, we have found the issue and fixed it.
image

Could you try it again? Thanks!

**Comment by yorek** [comment]
All working now, thanks! Merging the PR now.

**Pull request merged**